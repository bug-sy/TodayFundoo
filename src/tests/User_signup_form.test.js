import React from 'react';
import { shallow } from 'enzyme';
import UserSignup from '../components/User_signup_form';
import TextField from "@material-ui/core/TextField";

describe('User Signup Form', () => {


  it("should check for onChange method (3)", () => {
    // const wrapper = shallow(<Comment onChange={}/>)
  

    const component = shallow(
      <UserSignup  commentBody={"test"} />
    );
    const form = component
      .find(TextField)
      .at(0)
      .props()
      .onChange({target: {
        name: 'firstname',
        value: 'rakesh.sahu694@gmail.com'
      }});
      
      // th
      expect(component.state('firstnameTest')).toEqual('rakesh.sahu694@gmail.com');
  });

  it("should check for onChange method (3)", () => {
    const component = shallow(
      <UserSignup  commentBody={"test"} />
    );
    const form = component
      .find(TextField)
      .at(1)
      .props()
      .onChange({target: {
        name: 'lastname',
        value: 'abcdefg'
      }});
      
      // th
      expect(component.state('lastnameTest')).toEqual('abcdefg');
   });

   it("should check for onChange method (3)", () => {
    const component = shallow(
      <UserSignup  commentBody={"test"} />
    );
    const form = component
      .find(TextField)
      .at(2)
      .props()
      .onChange({target: {
        name: 'email',
        value: 'abcdefg'
      }});
      
      // th
      expect(component.state('emailTest')).toEqual('abcdefg');
   });
   it("should check for onChange method (3)", () => {
    const component = shallow(
      <UserSignup  commentBody={"test"} />
    );
    const form = component
      .find(TextField)
      .at(3)
      .props()
      .onChange({target: {
        name: 'password',
        value: 'abcdefg'
      }});
      
      // th
      expect(component.state('passwordTest')).toEqual('abcdefg');
   });

   it("should check for onChange method (3)", () => {
    const component = shallow(
      <UserSignup  commentBody={"test"} />
    );
    const form = component
      .find(TextField)
      .at(4)
      .props()
      .onChange({target: {
        name: 'confirmPassword',
        value: 'abcdefg'
      }});
      
      // th
      expect(component.state('confirmPasswordTest')).toEqual('abcdefg');
   });
});