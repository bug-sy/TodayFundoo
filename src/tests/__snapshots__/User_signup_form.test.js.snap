// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Signup Form 1`] = `
<div
  class="sign-up-wrapper"
>
  <form>
    <div
      class="icon"
    >
      <div
        class="firstdiv"
      >
        <div>
          <h2
            class="sign_up_button"
          >
            Sign up for Account
          </h2>
          <div
            class="MuiFormControl-root MuiTextField-root firstname MuiFormControl-marginDense"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
            >
              First_name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    First_name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root lastname MuiFormControl-marginDense"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
            >
              Last name
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Last name
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root Email MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
              for="#"
              id="#-label"
            >
              Email
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="#"
                type="Email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <br />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            Email can contain period.
          </div>
        </div>
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
              for="#1"
              id="#1-label"
            >
              Password
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="#1"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
                data-shrink="false"
                for="#2"
                id="#2-label"
              >
                Confirm Password
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
              >
                <input
                  aria-invalid="false"
                  autocomplete="current-password"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                  id="#2"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Confirm Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Sign Up
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div>
        <img
          alt="Smiley face"
          class="image"
          src="https://ssl.gstatic.com/accounts/signup/glif/account.svg"
        />
      </div>
    </div>
  </form>
</div>
`;
